### DISCUSSION TITLE
config rule not being applied when `baseBranches` is set with multiple branches

### DISCUSSION THREAD

**Discussion by timja** [renovatebot/renovate#30604] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Config rule is:
```json
    {
      "matchPackageNames": ["net.jcip:jcip-annotations"],
      "matchDatasources": ["maven"],
      "enabled": false,
      "description": "maven-metadata.xml is missing for this really old package which is required by renovate"
    }
```

No `baseBranches` config and it works as expected:
https://github.com/timja-org/jenkins-renovate-master-only/blob/d370769fd6a077e2c039579e339868755f5c5799/.github/renovate.json

https://developer.mend.io/github/timja-org/jenkins-renovate-master-only
(unrelated warning is there, that is being sorted by removing the dep in a separate PR)
<img width="1082" alt="image" src="https://github.com/user-attachments/assets/a8117185-0b84-4117-88f6-e4f115d3bda1">

`baseBranches` configured for master and stable-2.462:
https://github.com/timja-org/jenkins/blob/4b22341184899a297ba2f9cc0ff70d73c570987b/.github/renovate.json

warning for both dependencies:
<img width="1183" alt="image" src="https://github.com/user-attachments/assets/fddbb12d-e4a1-4f89-8d2e-1e273117e31e">

difference is these are removed:
```json
{
"baseBranches": ["master", "stable-2.462"],
"packageRules": [
    {
      "matchBaseBranches": ["/stable-.+/"],
      "matchManagers": ["maven"],
      "labels": ["dependencies", "into-lts", "needs-justification"]
    },
    {
      "matchBaseBranches": ["/stable-.+/"],
      "matchManagers": ["custom.regex", "npm", "github-actions"],
      "enabled": false
    }
}
```

Note: I've also seen other weird behaviour with `baseBranches` wildcards and regexes in matchPackageNames didn't seem to work, I had to expand a couple of rules to a full package match (e.g. the springframework rule)

### Logs (if relevant)

<details><summary>Logs from no baseBranches</summary>

```
DEBUG: Dependency: net.jcip:jcip-annotations, is disabled

          {
            "datasource": "maven",
            "depName": "net.jcip:jcip-annotations",
            "currentValue": "1.0",
            "fileReplacePosition": 6376,
            "registryUrls": [
              "https://repo.jenkins-ci.org/public/",
              "https://repo.maven.apache.org/maven2"
            ],
            "depType": "compile",
            "updates": [],
            "packageName": "net.jcip:jcip-annotations",
            "skipReason": "disabled"
          },

```

</details>

<details><summary>Logs from baseBranches version</summary>

```

DEBUG: Dependency: net.jcip:jcip-annotations, is disabled
{
  "baseBranch": "master"
}

        {
            "currentValue": "1.0",
            "datasource": "maven",
            "depName": "net.jcip:jcip-annotations",
            "depType": "compile",
            "fileReplacePosition": 6379,
            "packageName": "net.jcip:jcip-annotations",
            "registryUrls": [
              "https://repo.jenkins-ci.org/public/",
              "https://repo.maven.apache.org/maven2"
            ],
            "skipReason": "disabled",
            "versioning": "maven",
            "warnings": [
              {
                "message": "Failed to look up maven package net.jcip:jcip-annotations",
                "topic": "net.jcip:jcip-annotations"
              }
            ],
            "updates": []
          },
WARN: Package lookup failures (branch="renovate/stable-2.462-args4j-args4j-2.x")
{
  "baseBranch": "stable-2.462"
  "warnings": [
    "Failed to look up maven package net.jcip:jcip-annotations",
    "Failed to look up maven package org.maven.ide.eclipse:lifecycle-mapping"
  ]
  "files": [
    "bom/pom.xml",
    "pom.xml"
  ]

```

</details>

**Comment by viceice** [comment]
I think `matchDatasources` rules are applied after lookup, so you probably need to use `matchManagers` instead

**Reply by timja** [reply]
I can try but it works with the no default branch version

**Reply by timja** [reply]
Didn't work:

<img width="1153" alt="image" src="https://github.com/user-attachments/assets/8a868b1e-8ee5-4389-970f-9fdb670c9905">

https://github.com/timja-org/jenkins/commit/378eef52149eea08848d3f6386f93a06f392f51e

**Reply by viceice** [reply]
@rarkins an idea why it's still looked up? I'm out of ideas 🤔

**Reply by rarkins** [reply]
I'm having a little trouble following. Which rule is not working if baseBranches are configured?

**Reply by timja** [reply]
```json
    {
      "matchPackageNames": ["net.jcip:jcip-annotations"],
      "matchDatasources": ["maven"],
      "enabled": false,
      "description": "maven-metadata.xml is missing for this really old package which is required by renovate"
    }
```

**Reply by rarkins** [reply]
When I do a dry run lookup on this repo I now see this:

```
               {
                 "datasource": "maven",
                 "depName": "net.jcip:jcip-annotations",
                 "currentValue": "1.0",
                 "fileReplacePosition": 6378,
                 "registryUrls": [
                   "https://repo.jenkins-ci.org/public/",
                   "https://repo.maven.apache.org/maven2"
                 ],
                 "depType": "compile",
                 "updates": [],
                 "packageName": "net.jcip:jcip-annotations",
                 "skipReason": "disabled"
               },
```

I think this is correct/expected. Are you seeing different in the app logs?
